src/ca/weblite/codename1/io/tar/TarEntry.java

```diff
@@ -138,7 +138,7 @@ public void setSize(long size) {
      */
     public boolean isDirectory() {
         if (this.file != null)
-            return FileSystemStorage.getInstance().isDirectory(this.file);
+            return FileSystemStorage.getInstance().isDirectory(FileSystemStorage.getInstance().getAppHomePath() + "/" + this.file);

```

Comment:
You may be right.  This was the only fix I could find for the error in my complex application, but when I made a minimal test case the original code didn't show the issue...  I think this can be parked for now and I'll see if it occurs in other projects in the future.